### PR TITLE
NUX Signup: Use site topic redux state for site creation

### PR DESCRIPTION
**Pull request by southp**
#### Changes proposed in this Pull Request

This PR makes `SignupActions.createSiteWithCart()` use `signup.steps.siteTopic` state instead of `signup.steps.survey.vertical` when it is available. i.e. when we put the site topic picking step into the flow, it will outweigh the survey form from `about` step.

#### Testing instructions

Run `npm run test-client step-actions`.

### CODE REVIEW

File: client/lib/signup/step-actions.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ import { getDesignType } from 'state/signup/steps/design-type/selectors';
 import { getSiteTitle } from 'state/signup/steps/site-title/selectors';
 import { getSurveyVertical, getSurveySiteType } from 'state/signup/steps/survey/selectors';
 import { getSiteType } from 'state/signup/steps/site-type/selectors';
+import getSignupStepsSiteTopic from 'state/selectors/get-signup-steps-site-topic';
```
Comment (michaeldcain) [inline client/lib/signup/step-actions.js]: I missed this in #28164, but can we move this selector (and its test) to `state/signup/steps/site-topic/selectors`, just to follow the previous Signup redux conventions (and declutter `state/selectors/`?

You can handle it in a separate PR if you'd like.

File: client/lib/signup/step-actions.js [hunk 2]
```diff
@@ -102,6 +103,12 @@ export function createSiteOrDomain( callback, dependencies, data, reduxStore ) {
 	}
 }
 
+// We are experimenting making site topic a separate step from the survey.
+// Once we've decided to fully move away from the survey form, we can just keep the site topic here.
+function getSiteVertical( state ) {
+	return getSignupStepsSiteTopic( state ) || getSurveyVertical( state ).trim();
```
Comment (michaeldcain) [inline client/lib/signup/step-actions.js]: I know it's trimmed at the input, but I'm wondering if there is any risk by not trimming it again before sending to the endpoint?

File: client/lib/signup/step-actions.js (hunk 1 above)
Comment (southp) [reply client/lib/signup/step-actions.js]: Moved in 3c92d33.

File: client/lib/signup/step-actions.js (hunk 2 above)
Comment (southp) [reply client/lib/signup/step-actions.js]: I always honor defensive programming principles so why not. Trimming the return value as a whole in 0532848.

**Comment by sirreal** [comment]
Client tests on master started failing after this PR was merged: https://circleci.com/gh/Automattic/wp-calypso/133921#build-timing/containers/3

It's strange that the tests all pass on this branch. I'm going to track it down because all new PRs and current master tests are failing.

**Comment by sirreal** [comment]
#28505 [added `abtest`](https://github.com/Automattic/wp-calypso/pull/28505/files#diff-160d380c201b4db2d947c495e877a98aR17) to step-actions which uses `window` in an unguarded way.

This PR added a test for step actions. That passes on this branch (created before abtest was included), but when the two changes land on master together 💥 

I've prepared a fix which mocks the analytics lib in the new test:

#28579

**Pull request merged**